src/selection_stream/selection_stream_processor.ts

```diff
@@ -131,6 +132,10 @@ export class SelectionStreamProcessorImpl implements SelectionStreamProcessor {
     return this.modifyAnchor(anchor, "overrideSelection", "ZonesSelected");
   }
 
+  selectZoneAndKeepViewport(anchor: AnchorZone): DispatchResult {
```

Comment:
@rrahir what do you think ?
I think we must admit the selection processor cannot completely be decoupled from viewport (the various modes introduced just for the sake of handling them differently in the viewport already hinted it)

Reply:
Tought I posted an answer, apparently never did, sorry. 
It was kind of admitted and they're not that decoupled seeing as viewport needs to listen to everything already.

I agree that we cannot work around it anymore even though the split made sense from a functional pov at the start. But I don't know how to solve that issue :/

Reply:
the processor could expose its dependency and the methods have a 'scrollIntoView' argument ? that might alleviate the number of mode and types but also mean that the components would take control of what's displayed. That makes sense for the components, but maybe less for the plugins?